scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java

```diff
@@ -1,26 +1,50 @@
 
 package com.sap.scimono.entity.schema.validation;
 
-import com.sap.scimono.SCIMApplication;
+import static com.sap.scimono.entity.schema.validation.CustomInputValidator.Type.GROUP_ID;
+import static com.sap.scimono.entity.schema.validation.CustomInputValidator.Type.RESOURCE_ID;
+import static com.sap.scimono.entity.schema.validation.CustomInputValidator.Type.USER_ID;
+
+import java.util.Map;
 
 import javax.validation.ConstraintValidator;
 import javax.validation.ConstraintValidatorContext;
 import javax.ws.rs.core.Application;
 import javax.ws.rs.core.Context;
-import java.util.Map;
+import javax.ws.rs.core.UriInfo;
 
-import static com.sap.scimono.entity.schema.validation.CustomInputValidator.Type.RESOURCE_ID;
+import com.sap.scimono.SCIMApplication;
+import com.sap.scimono.api.API;
 
 public class IdValidator implements ConstraintValidator<ValidId, Object> {
   private Map<CustomInputValidator.Type, CustomInputValidator> customValidators;
+  private UriInfo uriInfo;
+
+  public IdValidator(@Context Application application, @Context UriInfo uriInfo) {
+    customValidators = SCIMApplication.from(application).getCustomConstraintValidators();
+    this.uriInfo = uriInfo;
 
-  public IdValidator(@Context Application application) {
-    this.customValidators = SCIMApplication.from(application).getCustomConstraintValidators();
   }
 
   @Override
   public boolean isValid(Object resourceId, ConstraintValidatorContext constraintValidatorContext) {
-    if(!customValidators.containsKey(RESOURCE_ID) || customValidators.get(RESOURCE_ID).isValid(resourceId)) {
+    if (uriInfo.getPath().startsWith(API.GROUPS) && customValidators.containsKey(GROUP_ID)) {
```

Comment:
Looks good, but let's see if we can avoid the nested if statements.

Reply:
Then the two nested if statements will become 4 if statements.
Should I change it?

Reply:
Maybe sth like this:

`@Override
  public boolean isValid(Object resourceId, ConstraintValidatorContext constraintValidatorContext) {
    if (customValidators.isEmpty()) {
      return true;
    }

    if (uriInfo.getPath().startsWith(API.GROUPS) && customValidators.containsKey(GROUP_ID) && customValidators.get(GROUP_ID).isValid(resourceId)) {
      return true;
    }

    if (uriInfo.getPath().startsWith(API.USERS) && customValidators.containsKey(USER_ID) && customValidators.get(USER_ID).isValid(resourceId)) {
      return true;
    }

    if (customValidators.containsKey(RESOURCE_ID) && customValidators.get(RESOURCE_ID).isValid(resourceId)) {
      return true;
    }

    ValidationUtil.interpolateErrorMessage(constraintValidatorContext, generateViolationMessage(resourceId));
    return false;
  }`

Reply:
Let's discuss it tomorrow